protocol/synthetix/contracts/storage/Market.sol

```diff
@@ -138,11 +138,16 @@ library Market {
          * @dev The maximum amount of market provided collateral, per type, that this market can deposit.
          */
         mapping(address => uint256) maximumDepositableD18;
-        uint32 minDelegateTime;
+        /**
+         * @dev Delegation/Undelegation frontrunning protection.
+         */
+        uint32 __unusedLegacyStorageSlot;
+        uint32 undelegateCollateralDelay;
+        uint32 undelegateCollateralWindow;
+        uint32 delegateCollateralDelay;
+        uint32 delegateCollateralWindow;
         uint32 __reservedForLater1;
         uint64 __reservedForLater2;
-        uint64 __reservedForLater3;
-        uint64 __reservedForLater4;
```

Comment:
Listing storage changes as this looks risky:

1. Deprecating `uint32  minDelegateTime` as `uint32 __unusedLegacyStorageSlot` (This should be marked with something like "DO NOT USE", because maybe it has data). ⚠️ 
2. Renaming `uint32 __reservedForLater1` to `uint32 undelegateCollateralDelay` ✅ 
3. Expanding slot `uint64 __reservedForLater3` to `uint32 undelegateCollateralWindow` & `uint32 delegateCollateralDelay` ✅
4. Renaming and **Resizing** `uint64 __reservedForLater4` to `uint32 delegateCollateralWindow`.

Please confirm @leomassazza, but this needs to add a `uint32 __reservedForLaterX` so it doesn't leak into the next slot for `uint256 minLiquidityRatioD18`